[Security Solution][Detection Engine] Only check read privileges of extant indices during rule execution

rylnd opened the pull request:
## Summary

This PR modifies the privilege-checking behavior during rule execution, restricting the indices against which we verify `read` access to only those that exist.

### Outstanding questions

- [x] Are there any backwards-compatibility/semver concerns with changing this behavior?
    * We discussed in which situations a user might reasonably be using the existing behavior, and determined those to be borderline. If we end up receiving feedback to the contrary, we can add back the old behavior as configuration.
- [x] Is the `IndexPatternsFetcher` an appropriate implementation to use for the existence checking?


### Steps to Review
1. Create a rule with a pattern including a non-existent index, e.g. `auditbeat-*,does-not-exist` 
2. Enable the rule, and observe no warning about e.g. missing read privileges for `does-not-exist`
3. (optional) Remove read access to `auditbeat-*`, or extend the pattern to include an existing index that the rule author cannot read
4. (optional) Observe a warning for the non-readable index


### Checklist

Delete any items that are not applicable to this PR.

- [ ] [Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html) was added for features that require explanation or tutorials
- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios

### For maintainers

- [ ] This was checked for breaking API changes and was [labeled appropriately](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

rylnd committed:
Only check read privileges of extant indices during rule execution

This is a WIP. Integration tests are failing because the `host_alias`
index is being listed as "DNE" by the getExistingIndices method.
Investigating.

rylnd committed:
Fix logical error by checking index existence with internal user

We can't use the current user to check whether indices exist, because
they might not be able to see all indices.

This fixes the above, and adds an appropriate integration test to cover
this particular case.

rylnd commented:
/ci

rylnd committed:
Merge branch 'main' into rylnd/bugfix/rule_warning_index_permission

rylnd commented:
/ci

rylnd committed:
Only check privileges if there are extant indices

This was previously calling checkPrivilegesFromEsClient even if there
were no indices to check, which resulted in an error response from ES
and broke things here.

There are integration tests around the "no existing indices" behavior,
which failed until we fixed this 👍.

rylnd commented:
/ci

rylnd committed:
Merge branch 'main' into rylnd/bugfix/rule_warning_index_permission

rylnd committed:
Fix tests, and add more

* Fixes an issue with the create_query_alert_type test which was relying
  on a code path that now short-circuits due to a lack of available
  indices. I've updated that test to follow the original path by mocking
  an additional ES call, as well as added a test for the
  short-circuiting behavior described.
* I also added an integration test around the short-circuiting behavior,
  to capture it moving forward.

rylnd commented:
Alright, this one ended up getting a little hairy, but I think we're good now. 

## Previous behavior
We previously checked permissions of nonexistent indices. Nonexistent indices would not have "read" privileges, so we would issue a warning saying as much. And because of the "no read privileges" warning, we [would not check](https://github.com/elastic/kibana/blob/f87a348/x-pack/plugins/security_solution/server/lib/detection_engine/rule_types/create_security_rule_type_wrapper.ts#L269-L298) whether _any_ indices had timestamp fields. And because we did not check timestamp fields, we would not set skipExecution to true (as the result of timestamp-field-checking implicitly checking for the existence of indices), and so the rule would execute on absent indices.

Unless it's somehow possible to get back a `read` permission on a non-existent index (short-circuit due to an admin role or something like that?), I don't think rule execution would ever have short-circuited due to no indices being present.

## New behavior
If we only check permissions of existing indices, we don't get the "no read privilege" warning for the nonexistent indices, so we proceed to check if the specified have timestamp fields.  In the case of no indices existing, this causes the rule execution to short-circuit.

It took me a while to wrap my head around this behavior, but we've now got tests confirming as much.

rylnd commented:
/ci

rylnd committed:
Merge branch 'main' into rylnd/bugfix/rule_warning_index_permission

rylnd commented:
/ci

elasticmachine commented:
Pinging @elastic/security-solution (Team: SecuritySolution)

elasticmachine commented:
Pinging @elastic/security-detection-engine (Team:Detection Engine)

rylnd committed:
Merge branch 'main' into rylnd/bugfix/rule_warning_index_permission

nkhristinin reviewed (dismissed):
Tested locally, LGTM!

nkhristinin commented:
As we discussed it online, I create this issue about show warning for indices which not exist - https://github.com/elastic/kibana/issues/180865

rylnd commented:
Closing as this will not address the underlying issue. If useful in the future, we can resurrect parts of this.

rylnd committed:
Merge branch 'main' into rylnd/bugfix/rule_warning_index_permission

rylnd committed:
Update constructor for new API

These arguments changed in
https://github.com/elastic/kibana/pull/180741.

It looks like the default value for `allowNoIndices` (which was the
`true` positional arg we were passing previously passing) now defaults
to `false`, but in that PR I saw other equivalent calls being changed to
the form we see here. It seems to work, but I need to validate that the
behavior here really isn't changed.

approksiu commented:
Thanks @rylnd! Let's proceed with adding the warnings for all cases of missing indices and privileges:

Indices matching index patterns/data view definitions exist:
1. All exist -> no warning
2. Some exist	-> "Indexes matching ______ not found."
3. None exist	-> "No indices matching index patterns in this rule's definition found." ( I assume we have the message for this failure already?)
	
Indeces matching index patterns/data view definitions are readable:
1. All readable -> no warning
2. Some can be read -> "This rule does not have required privileges to the following index patterns: _____"
3. None can be read -> "This rule does not have required privileges to the index patterns in this rule's definition" (Or do we have an error message already?)

rylnd committed:
Warn if some specified index patterns do not contain matching indexes

* Changes similar "warning" utilities to have a common(ish) interface
* Updates some GLOBAL RULE LOGIC to carry through earlier warning
  messages that would have previously been overwritten
* Adds a single test for the current case

I'm a little wary of adding the `||=` logic to the rule executor, as it
almost certainly changes behavior somewhere. We'll see if any existing tests agree.

Reading through the code I'm trying to understand how this is _supposed_
to work at a high level; what is the relationship between these
different warnings? Some seem to be collected in groups, while others
are overwritten by subsequent checks. The latter may be bugs, I'll need
to discuss with the rest of the team.

rylnd committed:
Merge branch 'main' into rylnd/bugfix/rule_warning_index_permission

rylnd committed:
Revert changes around "wrote warning" logic inside rule executor

We don't really want/need to change this behavior, we simply want to
write the partial failure/message to inform the user, which
`warnIfUnmatchedIndexPatterns` already does.

rylnd commented:
/ci

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/220728)
* Commit: 25b3558979c8e85b825eabf78ecd1572f0e494c4

### Failed CI Steps
* [Investigations - Security Solution Cypress Tests #4](https://buildkite.com/elastic/kibana-pull-request/builds/220728#01909e09-b9bc-4b1e-bed1-1b29e6ee2ce3)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/220728#01909e09-b9bc-4b1e-bed1-1b29e6ee2ce3) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/220728/jobs/01909e09-b9bc-4b1e-bed1-1b29e6ee2ce3/artifacts/01909e34-4bbd-4a49-8b25-fb086333aceb) Investigations - Security Solution Cypress Tests #<span></span>4 / Alerts generated by building block rules Alerts should be visible on the Rule Detail page and not visible on the Overview page Alerts should be visible on the Rule Detail page and not visible on the Overview page

cc @rylnd

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/220728","number":220728,"commit":"25b3558979c8e85b825eabf78ecd1572f0e494c4"}],"number":220728}
buildkite-pr-comment-->

rylnd committed:
Merge branch 'main' into rylnd/bugfix/rule_warning_index_permission

rylnd commented:
/ci

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/220769)
* Commit: 7feceff34bef68d7283afd6897a6df7d44f5f691

### Failed CI Steps
* [Investigations - Security Solution Cypress Tests #4](https://buildkite.com/elastic/kibana-pull-request/builds/220769#01909fae-c663-472b-a06c-dcd076d7dd8b)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/220769#01909fae-c663-472b-a06c-dcd076d7dd8b) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/220769/jobs/01909fae-c663-472b-a06c-dcd076d7dd8b/artifacts/01909fd6-ea9f-497c-bbd4-bdc9bf2e47ad) Investigations - Security Solution Cypress Tests #<span></span>4 / Alerts generated by building block rules Alerts should be visible on the Rule Detail page and not visible on the Overview page Alerts should be visible on the Rule Detail page and not visible on the Overview page

### History
* :broken_heart: [Build #220728](https://buildkite.com/elastic/kibana-pull-request/builds/220728) failed 25b3558979c8e85b825eabf78ecd1572f0e494c4

cc @rylnd

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/220769","number":220769,"commit":"7feceff34bef68d7283afd6897a6df7d44f5f691"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/220728","number":220728,"commit":"25b3558979c8e85b825eabf78ecd1572f0e494c4"}],"number":220769}
buildkite-pr-comment-->

rylnd committed:
Merge branch 'main' into rylnd/bugfix/rule_warning_index_permission

 Conflicts:
	x-pack/solutions/security/plugins/security_solution/server/lib/detection_engine/rule_types/create_security_rule_type_wrapper.ts
	x-pack/solutions/security/plugins/security_solution/server/lib/detection_engine/rule_types/utils/utils.ts

rylnd committed:
Remove unused helper import

rylnd commented:
/ci

rylnd commented:
Hey @approksiu, I'm circling back on this PR now. I've just updated this branch with `main` and resolved conflicts, and awaiting CI to report status. I'm still gathering context (aka "remembering" 😅) and validating that the current behavior is as specified, but AFAIK there were no blockers to completing this work. Will update appropriately.

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/265378)
* Commit: 4ec0851b070591af5b0d8c1acc78e72e67ae9ae2

### Failed CI Steps
* [Defend Workflows Cypress Tests #2](https://buildkite.com/elastic/kibana-pull-request/builds/265378#019447b5-4c34-4895-867a-dc8db09cc5a8)
* [Defend Workflows Cypress Tests #11](https://buildkite.com/elastic/kibana-pull-request/builds/265378#019447b5-4c37-4b38-97c3-05c11bd3e35d)
* [Defend Workflows Cypress Tests on Serverless #10](https://buildkite.com/elastic/kibana-pull-request/builds/265378#019447b5-4c3d-44a2-9758-33e90855d362)

### History
* :broken_heart: [Build #220769](https://buildkite.com/elastic/kibana-pull-request/builds/220769) failed 7feceff34bef68d7283afd6897a6df7d44f5f691
* :broken_heart: [Build #220728](https://buildkite.com/elastic/kibana-pull-request/builds/220728) failed 25b3558979c8e85b825eabf78ecd1572f0e494c4

cc @rylnd

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/265378","number":265378,"commit":"4ec0851b070591af5b0d8c1acc78e72e67ae9ae2"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/220769","number":220769,"commit":"7feceff34bef68d7283afd6897a6df7d44f5f691"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/220728","number":220728,"commit":"25b3558979c8e85b825eabf78ecd1572f0e494c4"}],"number":265378}
buildkite-pr-comment-->

rylnd committed:
Fix test failing due to upstream changes

The short-ciruiting behavior was removed in #186096, and so this test
case no longer applies. Instead it's replaced by the current behavior,
which is: to execute, but issue a warning with the appropriate message.

rylnd commented:
@approksiu regarding your [previous comment](https://github.com/elastic/kibana/pull/177658#issuecomment-2144538820):  I've verified that the appropriate warnings are issued both for lack of indices, and lack of read permissions, as specified. The one exception is that we do not special case the "no indices can be read" situation, but instead use the same message as the "some indices can't be read" case (which is demonstrated by [this test](https://github.com/elastic/kibana/pull/177658/files#diff-1edd580040b42ac9a2bf8ccb2cc457027246a49f735b560be4af9e33f8e07c3aR117) and [its assertion](https://github.com/elastic/kibana/pull/177658/files#diff-1edd580040b42ac9a2bf8ccb2cc457027246a49f735b560be4af9e33f8e07c3aR165).

I believe the remaining failures are all due generally to the broad change in behavior introduced here: whereas previously were were setting a "successful" rule execution in these scenarios, we now have a "partial failure" result instead (cc @marshallmain and @yctercero on this point).

If you can please verify that the above is acceptable/expected, I'll fix the outstanding tests and then this should be good to merge.

approksiu commented:
@rylnd thanks for looking into this. 
> The one exception is that we do not special case the "no indices can be read" situation, but instead use the same message as the "some indices can't be read" case (which is demonstrated by [this test](https://github.com/elastic/kibana/pull/177658/files#diff-1edd580040b42ac9a2bf8ccb2cc457027246a49f735b560be4af9e33f8e07c3aR117) and [its assertion](https://github.com/elastic/kibana/pull/177658/files#diff-1edd580040b42ac9a2bf8ccb2cc457027246a49f735b560be4af9e33f8e07c3aR165).

Let's go with this approach, and open a ticket to address the "no idexes accessible"-> failed rule (not a partial failure) error.

> I believe the remaining failures are all due generally to the broad change in behavior introduced here: whereas previously were were setting a "successful" rule execution in these scenarios, we now have a "partial failure" result instead (cc @marshallmain and @yctercero on this point).

Which failures do you mean?

Thanks!

rylnd committed:
Move status-writing responsibility out of helper function

In order to better conform with the changes made in #186096 (which was
created after but merged before this branch), we do not
write a rule status from within the check for unmatched index patterns,
but instead return a warning message that the caller then handles.

This allows the rule executor to decide what status should be written,
and without this change we were writing both a partial_failure (due to
the unmatched pattern), and also a success (due to the executor not
knowing about the warning).

rylnd committed:
Update Rule Management tests to reflect latest behavior

We now record a rule as a partial_failure in the case where _some_, but
not _all_, of a rule's specified index patterns match an existing index.
Previously, this situation would be treated as a successful execution.

rylnd commented:
> Which failures do you mean?

@approksiu I was referring to the failures in [this build](https://buildkite.com/elastic/kibana-pull-request/builds/265378), but there's some noise there as well (with outdated/unrelated failures). 

Besides the changes summarized [above](https://github.com/elastic/kibana/pull/177658#issuecomment-2038213388), one  major behavioral change in this PR is shown in this commit: https://github.com/elastic/kibana/pull/177658/commits/9b0f6c74cee2a870d85b02d59ef094a080945367.

> We now record a rule as a partial_failure in the case where _some_, but
not _all_, of a rule's specified index patterns match an existing index.
Previously, this situation would be treated as a successful execution.

I bring this up because:

1. This will impact a significant number of rules, certainly our prebuilt rules, which use a broad index pattern to catch most/all integrations.
2. This new behavior is a natural consequence of reporting on this situation; as far as I know we don't have a mechanism to report a warning _without_ writing a `partial_failure`.

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/266248)
* Commit: 9b0f6c74cee2a870d85b02d59ef094a080945367

### Failed CI Steps
* [Serverless Investigations - Security Solution Cypress Tests #6](https://buildkite.com/elastic/kibana-pull-request/builds/266248#01945242-531c-4737-847c-a62653295600)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/266248#01945242-531c-4737-847c-a62653295600) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/266248/jobs/01945242-531c-4737-847c-a62653295600/artifacts/01945277-4ba2-4fe9-9cd9-b54105a85e16) Serverless Investigations - Security Solution Cypress Tests #<span></span>6 / Alerts generated by building block rules Alerts should be visible on the Rule Detail page and not visible on the Overview page Alerts should be visible on the Rule Detail page and not visible on the Overview page

### History
* :broken_heart: [Build #265378](https://buildkite.com/elastic/kibana-pull-request/builds/265378) failed 4ec0851b070591af5b0d8c1acc78e72e67ae9ae2
* :broken_heart: [Build #220769](https://buildkite.com/elastic/kibana-pull-request/builds/220769) failed 7feceff34bef68d7283afd6897a6df7d44f5f691
* :broken_heart: [Build #220728](https://buildkite.com/elastic/kibana-pull-request/builds/220728) failed 25b3558979c8e85b825eabf78ecd1572f0e494c4

cc @rylnd

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/266248","number":266248,"commit":"9b0f6c74cee2a870d85b02d59ef094a080945367"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/265378","number":265378,"commit":"4ec0851b070591af5b0d8c1acc78e72e67ae9ae2"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/220769","number":220769,"commit":"7feceff34bef68d7283afd6897a6df7d44f5f691"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/220728","number":220728,"commit":"25b3558979c8e85b825eabf78ecd1572f0e494c4"}],"number":266248}
buildkite-pr-comment-->

rylnd committed:
Do not warn when index pattern contains unmatched indices

This reverts this behavior for now, as we have #180865 to discuss/decide
on the best option moving forward.

rylnd committed:
Revert variable rename

This was part of a larger commit, but this file is now untouched on this
branch, so these changes are mostly just noise at this point.

rylnd committed:
Update unit tests to document current behavior

rylnd committed:
Remove tests exercising old behavior

This behavior was removed in a previous commit. However, I'm keeping the
`context` and index definition since the scenario still exists, and we
do want to eventually address it.

rylnd commented:
/ci

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/266794)
* Commit: 039697f0311be714b65f24ad1bb30a7b667725e2

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/266794#0194633f-9bbd-418f-993a-16311dae3d84)

### History
* :broken_heart: [Build #266248](https://buildkite.com/elastic/kibana-pull-request/builds/266248) failed 9b0f6c74cee2a870d85b02d59ef094a080945367
* :broken_heart: [Build #265378](https://buildkite.com/elastic/kibana-pull-request/builds/265378) failed 4ec0851b070591af5b0d8c1acc78e72e67ae9ae2
* :broken_heart: [Build #220769](https://buildkite.com/elastic/kibana-pull-request/builds/220769) failed 7feceff34bef68d7283afd6897a6df7d44f5f691
* :broken_heart: [Build #220728](https://buildkite.com/elastic/kibana-pull-request/builds/220728) failed 25b3558979c8e85b825eabf78ecd1572f0e494c4

cc @rylnd

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/266794","number":266794,"commit":"039697f0311be714b65f24ad1bb30a7b667725e2"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/266248","number":266248,"commit":"9b0f6c74cee2a870d85b02d59ef094a080945367"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/265378","number":265378,"commit":"4ec0851b070591af5b0d8c1acc78e72e67ae9ae2"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/220769","number":220769,"commit":"7feceff34bef68d7283afd6897a6df7d44f5f691"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/220728","number":220728,"commit":"25b3558979c8e85b825eabf78ecd1572f0e494c4"}],"number":266794}
buildkite-pr-comment-->

rylnd committed:
Remove unused variable in pending test context

rylnd committed:
Merge branch 'main' into rylnd/bugfix/rule_warning_index_permission

rylnd commented:
@approksiu @marshallmain @nkhristinin I've updated the branch to remove the behavior of https://github.com/elastic/kibana/issues/180865, so this is back to just what's described in the title: restricting the read privilege checking to existent indices. There are a few tests added here that demonstrate _existing_ partial_failure behavior, but none of that is added in this PR.

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/267140)
* Commit: 7c5777b5b067d0a6f07388fe72d481ec4a25d3af

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ✅ unchanged

### History
* :broken_heart: [Build #266794](https://buildkite.com/elastic/kibana-pull-request/builds/266794) failed 039697f0311be714b65f24ad1bb30a7b667725e2
* :broken_heart: [Build #266248](https://buildkite.com/elastic/kibana-pull-request/builds/266248) failed 9b0f6c74cee2a870d85b02d59ef094a080945367
* :broken_heart: [Build #265378](https://buildkite.com/elastic/kibana-pull-request/builds/265378) failed 4ec0851b070591af5b0d8c1acc78e72e67ae9ae2
* :broken_heart: [Build #220769](https://buildkite.com/elastic/kibana-pull-request/builds/220769) failed 7feceff34bef68d7283afd6897a6df7d44f5f691
* :broken_heart: [Build #220728](https://buildkite.com/elastic/kibana-pull-request/builds/220728) failed 25b3558979c8e85b825eabf78ecd1572f0e494c4

cc @rylnd

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/267140","number":267140,"commit":"7c5777b5b067d0a6f07388fe72d481ec4a25d3af"},{"buildStatus":{"state":"running","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/266794","number":266794,"commit":"039697f0311be714b65f24ad1bb30a7b667725e2"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/266248","number":266248,"commit":"9b0f6c74cee2a870d85b02d59ef094a080945367"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/265378","number":265378,"commit":"4ec0851b070591af5b0d8c1acc78e72e67ae9ae2"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/220769","number":220769,"commit":"7feceff34bef68d7283afd6897a6df7d44f5f691"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/220728","number":220728,"commit":"25b3558979c8e85b825eabf78ecd1572f0e494c4"}],"number":267140}
buildkite-pr-comment-->

rylnd committed:
Merge branch 'main' into rylnd/bugfix/rule_warning_index_permission

dplumlee reviewed (approved):
Rule management changes lgtm

nkhristinin reviewed (approved):
Tested locally, LGTM!

yctercero committed:
Merge branch 'main' into rylnd/bugfix/rule_warning_index_permission

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/268983)
* Commit: c7f31e383b353c2f3a9bc46efbd355f739645dbe

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ✅ unchanged

### History
* :green_heart: [Build #267140](https://buildkite.com/elastic/kibana-pull-request/builds/267140) succeeded 7c5777b5b067d0a6f07388fe72d481ec4a25d3af
* :broken_heart: [Build #266794](https://buildkite.com/elastic/kibana-pull-request/builds/266794) failed 039697f0311be714b65f24ad1bb30a7b667725e2
* :broken_heart: [Build #266248](https://buildkite.com/elastic/kibana-pull-request/builds/266248) failed 9b0f6c74cee2a870d85b02d59ef094a080945367
* :broken_heart: [Build #265378](https://buildkite.com/elastic/kibana-pull-request/builds/265378) failed 4ec0851b070591af5b0d8c1acc78e72e67ae9ae2
* :broken_heart: [Build #220769](https://buildkite.com/elastic/kibana-pull-request/builds/220769) failed 7feceff34bef68d7283afd6897a6df7d44f5f691
* :broken_heart: [Build #220728](https://buildkite.com/elastic/kibana-pull-request/builds/220728) failed 25b3558979c8e85b825eabf78ecd1572f0e494c4

cc @rylnd

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/268983","number":268983,"commit":"c7f31e383b353c2f3a9bc46efbd355f739645dbe"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/267140","number":267140,"commit":"7c5777b5b067d0a6f07388fe72d481ec4a25d3af"},{"buildStatus":{"state":"running","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/266794","number":266794,"commit":"039697f0311be714b65f24ad1bb30a7b667725e2"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/266248","number":266248,"commit":"9b0f6c74cee2a870d85b02d59ef094a080945367"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/265378","number":265378,"commit":"4ec0851b070591af5b0d8c1acc78e72e67ae9ae2"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/220769","number":220769,"commit":"7feceff34bef68d7283afd6897a6df7d44f5f691"}],"number":268983}
buildkite-pr-comment-->
